cmd/krel/cmd/announce_build.go

```diff
@@ -0,0 +1,264 @@
+	)
+
+	buildReleaseAnnounceCmd.PersistentFlags().StringVarP(
+		&buildReleaseAnnounceOpts.changelogHTML,
+		changeLogHTMLFlag,
+		"",
+		"",
+		"contents of the changelog",
+	)
+
+	buildAnnounceCmd.PersistentFlags().StringVarP(
+		&buildAnnounceOpts.workDir,
+		workDirFlag,
+		"",
+		"",
+		"working directory to store the annoucement files",
+	)
+
+	if err := buildAnnounceCmd.MarkPersistentFlagRequired(workDirFlag); err != nil {
+		logrus.Fatal(err)
+	}
+
+	// Check flags for krel announcement build branch command
+	if err := buildBranchAnnounceCmd.MarkPersistentFlagRequired(branchFlag); err != nil {
+		logrus.Fatal(err)
+	}
+
+	// Check flags for krel announcement build release command
+	if err := buildReleaseAnnounceCmd.MarkPersistentFlagRequired(changeLogFilePathFlag); err != nil {
+		logrus.Fatal(err)
+	}
+
+	if err := buildReleaseAnnounceCmd.MarkPersistentFlagRequired(changeLogHTMLFlag); err != nil {
+		logrus.Fatal(err)
+	}
+
+	buildAnnounceCmd.AddCommand(buildBranchAnnounceCmd)
+	buildAnnounceCmd.AddCommand(buildReleaseAnnounceCmd)
+	announceCmd.AddCommand(buildAnnounceCmd)
+}
+
+// runBuildBranchAnnounce build the announcement file when creating the Kubernetes release branch
+func runBuildBranchAnnounce(opts *buildBranchAnnounceOptions, buildOpts *buildAnnounceOptions) error {
+	logrus.Info("Building release announcement for branch creation")
+
+	t, err := template.New("announcement-branch").Parse(branchCreationMsg)
+	if err != nil {
+		return err
+	}
+	annoucement := bytes.Buffer{}
+	if err := t.Execute(&annoucement, struct {
+		Branch string
+	}{opts.branch}); err != nil {
+		return errors.Wrap(err, "generating the announcement html file")
+	}
+
+	announcementSubject := fmt.Sprintf("Kubernetes %s branch has been created", opts.branch)
+	return buildOpts.saveAnnouncement(announcementSubject, annoucement)
+}
+
+// runBuildReleaseAnnounce build the announcement file when creating a new Kubernetes release
+func runBuildReleaseAnnounce(opts *buildReleaseAnnounceOptions, buildOpts *buildAnnounceOptions, announceOpts *announceOptions) error {
+	if err := announceOpts.Validate(); err != nil {
+		return errors.Wrap(err, "validating annoucement send options")
+	}
+
+	logrus.Info("Building release announcement for new release")
+
+	t, err := template.New("announcement-release").Parse(releaseAnnouncementMsg)
+	if err != nil {
+		return err
+	}
+
+	goVersion, err := getGoVersion()
+	if err != nil {
+		return err
+	}
+
+	changelogHTML, err := ioutil.ReadFile(opts.changelogHTML)
+	if err != nil {
+		return err
+	}
+
+	annoucement := bytes.Buffer{}
+	if err := t.Execute(&annoucement, struct {
+		Tag               string
+		StrippedTag       string
+		GoVersion         string
+		ChangelogFilePath string
+		ChangelogFileName string
+		ChangelogHTML     string
+	}{
+		announceOpts.tag,
+		strings.ReplaceAll(announceOpts.tag, ".", ""),
+		goVersion,
+		opts.changelogFilePath,
+		strings.Split(opts.changelogFilePath, "/")[1],
```

Comment:
Can we use something like [`filepath.Base`](https://golang.org/pkg/path/filepath/#example_Base) here?

Reply:
yep, forgot that :(

Reply:
done